### PR TITLE
feat: add the pubsublite sink

### PR DESCRIPTION
**Pull request by palmere-google**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-pubsublite-flink/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -101,6 +101,16 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-streaming-java_2.11</artifactId>
+            <version>1.13.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-runtime_2.11</artifactId>
+            <version>1.13.0</version>
```
Comment (dpcollins-google) [inline pom.xml:112]: nit. add a property "flink.version" and use it everywhere you have 1.13.0. See pubsublite-beam-io for an example.

File: src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSink.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  @GuardedBy("this")
+  private transient BulkWaitPublisher<Tuple<T, Instant>> publisher;
+
+  public PubsubLiteSink(PubsubLiteSinkSettings<T> settings) {
+    this.settings = settings;
+  }
+
+  @Override
+  public synchronized void snapshotState(
+      org.apache.flink.runtime.state.FunctionSnapshotContext functionSnapshotContext)
```
Comment (dpcollins-google) [inline src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSink.java:41]: Why not import these?

File: src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  PublisherOptions getPublisherConfig() {
+    return PublisherOptions.create(topicPath());
+  }
+
+  @AutoValue.Builder
+  abstract static class Builder<InputT> {
+    public abstract Builder<InputT> setTopicPath(TopicPath value);
+
+    abstract Builder<InputT> setSerializationSchema(PubsubLiteSerializationSchema<InputT> value);
+
```
Comment (dpcollins-google) [inline src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java:49]: is this the proper user interface? If so, add comments.

File: src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import java.io.Serializable;
+
+@AutoValue
+public abstract class PubsubLiteSinkSettings<InputT> implements Serializable {
+  public static <InputT> PubsubLiteSinkSettings.Builder<InputT> builder(
+      PubsubLiteSerializationSchema<InputT> schema) {
+    return new AutoValue_PubsubLiteSinkSettings.Builder<InputT>().setSerializationSchema(schema);
+  }
+
+  public static PubsubLiteSinkSettings.Builder<Message> messagesBuilder() {
```
Comment (dpcollins-google) [inline src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java:31]: Just Builder

File: src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.pubsublite.Message;
+import com.google.cloud.pubsublite.TopicPath;
+import com.google.cloud.pubsublite.flink.sink.PublisherOptions;
+import java.io.Serializable;
+
+@AutoValue
+public abstract class PubsubLiteSinkSettings<InputT> implements Serializable {
+  public static <InputT> PubsubLiteSinkSettings.Builder<InputT> builder(
```
Comment (dpcollins-google) [inline src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java:26]: just Builder

**Review by dpcollins-google — APPROVED** [review]

File: src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSink.java (hunk 2 above)
Comment (palmere-google) [reply src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSink.java:41]: Done. Intellij generated them like this and I never changed them

File: pom.xml (hunk 1 above)
Comment (palmere-google) [reply pom.xml:112]: Will do in another pr, tianzi suggested that I do this for pubsublite too

File: src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java (hunk 3 above)
Comment (palmere-google) [reply src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java:49]: Added comments.

File: src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java (hunk 4 above)
Comment (palmere-google) [reply src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java:31]: Done, better, thanks

File: src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java (hunk 5 above)
Comment (palmere-google) [reply src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSinkSettings.java:26]: Done

**Pull request merged**